packages/switch/src/component.tsx

```diff
@@ -5,8 +5,34 @@ import { switchToggle as ccSwitch } from '@warp-ds/css/component-classes';
 
 import { SwitchProps } from './props.js';
 
-export function Switch({ id, value, onClick, 'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledBy, ...attrs }: SwitchProps) {
-  const switchFocus = 'focusable rounded-full';
+export function Switch({
+  id,
+  value,
+  disabled,
+  onClick,
+  'aria-label': ariaLabel,
+  'aria-labelledby': ariaLabelledBy,
+  ...attrs
+}: SwitchProps) {
+  const baseClasses = classNames([ccSwitch.base, disabled && ccSwitch.disabled]);
+
+  const trackClasses = classNames([
+    ccSwitch.track,
+    disabled && ccSwitch.trackDisabled,
+    value && !disabled ? ccSwitch.trackActive : ccSwitch.trackInactive,
+  ]);
+
+  const handleClasses = classNames([
+    ccSwitch.handle,
+    value && ccSwitch.handleSelected,
+    disabled ? ccSwitch.handleDisabled : ccSwitch.handleNotDisabled,
+  ]);
+
+  const handleClick = (e) => {
+    if (!disabled && onClick) {
+      onClick(e);
+    }
+  };
```

Comment:
We don't need this check because we have a `pointer-events-none` class set if the Switch is disabled, which makes the button not a target to any pointer events.

Reply:
the `pointer-events-none` seem to only work for the mouse, but you could still navigate with the keyboard and toggle the switch, so that's why I added this handleClick. :)

Reply:
Ah right! I'll remove this :)

Reply:
Updated in this commit: [fb41ad8](https://github.com/warp-ds/react/pull/270/commits/fb41ad85ab2fbdb3561e8692df4e89c07f63fcb0)